modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/transport/netty4/Netty4TransportMultiPortIntegrationIT.java

```diff
@@ -64,17 +66,28 @@ protected Settings nodeSettings(int nodeOrdinal, Settings otherSettings) {
         return builder.build();
     }
 
-    public void testThatTransportClientCanConnect() throws Exception {
+    public void testThatTransportClientCanConnect() {
         Settings settings = Settings.builder()
             .put("cluster.name", internalCluster().getClusterName())
             .put(NetworkModule.TRANSPORT_TYPE_KEY, Netty4Plugin.NETTY_TRANSPORT_NAME)
             .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toString())
             .build();
-        // we have to test all the ports that the data node might be bound to
         try (TransportClient transportClient = new MockTransportClient(settings, Arrays.asList(Netty4Plugin.class))) {
-            for (int i = 0; i <= NUMBER_OF_CLIENT_PORTS; i++) {
-                transportClient.addTransportAddress(new TransportAddress(InetAddress.getByName("127.0.0.1"), randomPort + i));
+            final List<TransportAddress> addresses = new ArrayList<>();
+            for (final Transport transport : internalCluster().getInstances(Transport.class)) {
+                for (final TransportAddress transportAddress : transport.boundAddress().boundAddresses()) {
+                    addresses.add(transportAddress);
+                }
+                for (final BoundTransportAddress profileAddresses : transport.profileBoundAddresses().values()) {
+                    for (final TransportAddress transportAddress : profileAddresses.boundAddresses()) {
+                        addresses.add(transportAddress);
+                    }
+                }
             }
+            logger.info("--> all addresses: {}", addresses);
```

Comment:
Is there any expectation for a minimum number of addresses that have been collected? Wondering if there's a sanity check that we are in fact testing some number of addresses.

Reply:
Yes every node must bind to at least one transport address, otherwise it wouldn't be able to join a cluster.